controllers/storagecluster/clusterclaims.go

```diff
@@ -0,0 +1,226 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+
+	"github.com/go-logr/logr"
+	extensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	RookCephMonSecretName = "rook-ceph-mon"
+	FsidKey               = "fsid"
+	OdfOperatorNamePrefix = "odf-operator"
+	ClusterClaimCRDName   = "clusterclaims.cluster.open-cluster-management.io"
+)
+
+var (
+	ClusterClaimGroup  = "odf"
+	OdfVersion         = fmt.Sprintf("version.%s.openshift.io", ClusterClaimGroup)
+	StorageSystemName  = fmt.Sprintf("storagesystemname.%s.openshift.io", ClusterClaimGroup)
+	StorageClusterName = fmt.Sprintf("storageclustername.%s.openshift.io", ClusterClaimGroup)
+	CephFsid           = fmt.Sprintf("cephfsid.%s.openshift.io", ClusterClaimGroup)
+)
+
+type ocsClusterClaim struct{}
+type ClusterClaimCreator struct {
+	Context        context.Context
+	Logger         logr.Logger
+	Client         client.Client
+	Values         map[string]string
+	StorageCluster *ocsv1.StorageCluster
+}
+
+func checkIfClusterClaimCRDExists(ctx context.Context, client client.Client) (bool, error) {
+	crd := extensionsv1.CustomResourceDefinition{}
+	err := client.Get(ctx, types.NamespacedName{Name: ClusterClaimCRDName}, &crd)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+
+	return true, nil
+}
+func (obj *ocsClusterClaim) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	ctx := context.TODO()
+	if crdExists, err := checkIfClusterClaimCRDExists(ctx, r.Client); !crdExists {
+		if err != nil {
+			r.Log.Info("An error has occured while fetching customresourcedefinition for %q", ClusterClaimCRDName)
```

Comment:
```suggestion
			r.Log.Error("An error has occured while fetching customresourcedefinition for %q", ClusterClaimCRDName)
```

Reply:
I am thinking of keeping it the same as the error is being returned. This will spam the same error message twice, once here and another by the reconciler